spring-xd-dirt/src/main/java/org/springframework/xd/dirt/stream/XDParser.java

```diff
@@ -43,4 +45,29 @@
 	 *         described by the DSL.
 	 */
 	List<ModuleDescriptor> parse(String name, String config, ParsingContext type);
+
+	public static class DefinitionOrException {
```

Comment:
Justifying architecture decisions based on "theoretical considerations" that are 99% likely NOT to ever happen is the #1 source of technical debt. The parser should parse a single stream definition IMO. Anything else needs to be a layer above that. If only used by FloController, we should not be adding yet more complexity to the parser itself. I would even prefer to see a decorated version of the parser that is used by FloController, and for now that could even be an inner class of FloController.

Thanks for the "tap:stream:foo" example since that clarifies the rationale for more than a cache/Map. However, that also aligns with a concern I have as we're trying to refactor the parser. I think we should have a separation between syntactic parsing and validation based on things like existing streams.